### ISSUE TITLE
awxkit doesn't expose job cancellation

### ISSUE THREAD

**Issue by phemmer** [ansible/awx#10201]
##### ISSUE TYPE
 - Feature Idea

##### SUMMARY
Add a command flag such that when using `awx job_templates launch --monitor`, and the command is interrupted (SIGINT/SIGTERM), the job is stopped before exiting.

The intent behind this is for use in version control CI systems. A CI job might launch an AWX job using `awx job_templates launch`, and then the CI job is cancelled (such as due to a push which obsoletes the current job), but the AWX job continues running. The admin might want to have the AWX job cancelled as well so that a new job can be kicked off (among other reasons).

There is a risk of course, such that handlers won't run and whatnot, but the admin can decide whether the risk is acceptable.

&nbsp;

Also note there's currently no way to hack around this, by launching job, getting ID, monitoring, and manually calling cancel. awxkit does not expose a cancellation call to the user.

**Comment by wenottingham** [comment]
>  awxkit does not expose a cancellation call to the user.

I think this is what we'd fix - having a ctrl-c send a cancel to the API doesn't seem like the right behavior. We'd want it to be an explicit command.

**Comment by phemmer** [comment]
The idea was to make it behave similar to running `ansible-playbook` locally. But it shouldn't be too difficult to write a wrapper script to handle the orchestration.

**Comment by Endemoniada** [comment]
> The idea was to make it behave similar to running ansible-playbook locally.

I have to agree with wenottingham. Being explicit is better than being implicit when doing something as potentially serious as stopping a running job, and having Ctrl-C send it automatically might get very confusing for those running it in CLI. Like, how would you then stop just _the monitoring_, without then also killing the running job?

There should _absolutely_ be a way to terminate a running job via API (and, frankly, I'm stunned this still isn't possible), but I cast my vote on it being a separate sub-command for the `jobs` endpoint.

**Comment by ricpatel** [comment]
Has there been any updates on this issue?
